### PR TITLE
Allow pacman to move in P3

### PR DESCRIPTION
**Pull request by PaLeroy**
Modification of the game engine files.

- Waiting for code review
- Waiting for README review
- Do not forget to update the dowlable archive

### CODE REVIEW

**Comment by PaLeroy** [comment]
We may consider to add a disclaimer stating that they can start working on the project and that only the game engine is subject to change if they are unknown bugs.

File: projects/project3/README.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ You are asked to answer the following questions:
 	- 3.d. - **2 points** - Discuss the effect of the ghost transition model parameter on its own behavior and on Pacman's belief state. Consider the two provided layouts. Motivate your answer by using your measures and the model itself. Use the default sensor variance.
 	- 3.e. - **2 points** - Discuss the effect of the sensor variance (as set through the `--sensorvariance` command line argument) on Pacman's belief state.
 	- 3.f. - **2 points** - How would you implement a stochastic Pacman controller to eat ghosts using only its current position, the set of legal actions and its current belief state?
+	- 3.d. - **BONUS TBD points** - Implement this stochastic controller in the `pacmanagent.py` file.
```
Comment (glouppe) [inline projects/project3/README.md:71]: 3 points?

File: projects/project3/README.md [hunk 2]
```diff
@@ -106,7 +107,7 @@ Besides the questions you're expected to answer, you will also be evaluated acco
 		- 0.3 point : at least 25% correct specifications.
 		- 0 point : less than 25% correct specifications.
 
-Note that your implementation might be tested on other layouts.
+Note that your implementation might be tested on other layouts with moving Pacman.
```
Comment (glouppe) [inline projects/project3/README.md:110]: again, not sure i understand

File: projects/project3/beliefstateagent.py [hunk 3]
```diff
@@ -113,7 +113,7 @@ def _record_metrics(self, belief_states, state):
 
     def get_action(self, state):
         """
-        Given a pacman game state, returns a legal move.
+        Given a pacman game state, returns a "belief state".
```
Comment (glouppe) [inline projects/project3/beliefstateagent.py:116]: remove quotes

File: projects/project3/beliefstateagent.py [hunk 4]
```diff
@@ -122,7 +122,7 @@ def get_action(self, state):
 
         Return:
         -------
-        - A legal move as defined in `game.Directions`.
+        - A "belief state".
```
Comment (glouppe) [inline projects/project3/beliefstateagent.py:125]: remove quotes

File: projects/project3/humanagent.py [hunk 5]
```diff
@@ -24,15 +24,17 @@ def __init__(self, args):
         self.lastMove = Directions.STOP
         self.keys = []
 
-    def get_action(self, state):
+    def get_action(self, state, belief_state):
         """
-        Given a pacman game state, returns a legal move.
+        Given a pacman game state and a "belief state",
```
Comment (glouppe) [inline projects/project3/humanagent.py:29]: remove quotes

File: projects/project3/humanagent.py [hunk 6]
```diff
@@ -24,15 +24,17 @@ def __init__(self, args):
...
+                returns a legal move.
 
         Arguments:
         ----------
         - `state`: the current game state. See FAQ and class
                    `pacman.GameState`.
-
+        - 'belief_state': a list of probability matrices.
```
Comment (glouppe) [inline projects/project3/humanagent.py:36]: use backticks (`)

File: projects/project3/pacmanagent.py [hunk 7]
```diff
@@ -13,17 +13,24 @@ def __init__(self, args):
         """
         self.args = args
 
-    def get_action(self, state):
+    def get_action(self, state, belief_state):
         """
-        Given a pacman game state, returns a legal move.
+        Given a pacman game state and a "belief state",
```
Comment (glouppe) [inline projects/project3/pacmanagent.py:18]: remove quotes

File: projects/project3/pacmanagent.py [hunk 8]
```diff
@@ -13,17 +13,24 @@ def __init__(self, args):
...
+                returns a legal move.
 
         Arguments:
         ----------
         - `state`: the current game state. See FAQ and class
                    `pacman.GameState`.
+        - 'belief_state': a list of probability matrices.
```
Comment (glouppe) [inline projects/project3/pacmanagent.py:25]: use backticks

File: projects/project3/README.md [hunk 9]
```diff
@@ -56,7 +56,7 @@ You are asked to answer the following questions:
  	- 2.a. - **4 points** - Implement the **Bayes filter** algorithm to compute Pacman's belief state. This should be done in the `update_belief_state` function of `bayesfilter.py`, following the template of `beliefstateagent.py`.
 		 - Your implementation must work with multiple ghosts (all running the same policy).
 		 - Pacman's belief state should eventually converge to an uncertainty area for each ghost.
-		 - Your filter should consider the Pacman position.
+		 - Your filter should consider the Pacman position as it can move.
```
[the remark this answers is not in the document]
Comment (glouppe) [reply projects/project3/README.md:59]: (Taking the position into account is anyway necessary)

[the remark this answers is not in the document]
Comment (glouppe) [reply projects/project3/README.md:59]: or to remove any doubt, rephrase as : "Your filter should consider the Pacman position, as Pacman may wander freely in the maze."

File: projects/project3/README.md (hunk 2 above)
Comment (glouppe) [reply projects/project3/README.md:110]: or rephrase as "Note that your implementation might be tested on other layouts, with Pacman moving abitrarily.

**Pull request merged**